src/OpenRiaServices.Hosting.AspNetCore/Framework/AspNetCore/OpenRiaServicesConfigurationBuilder.cs

```diff
@@ -19,19 +20,60 @@ internal OpenRiaServicesConfigurationBuilder(OpenRiaServicesEndpointDataSource d
             _typeIsService = typeIsService;
         }
 
+        public IEndpointConventionBuilder AddDomainService<T>() where T : DomainService
+        {
+            return AddDomainService(typeof(T));
+        }
+
         public IEndpointConventionBuilder AddDomainService(Type type)
         {
-            var longName = type.FullName.Replace('.', '-') + ".svc";
+            ArgumentNullException.ThrowIfNull(nameof(type));
 
             if (!_typeIsService.IsService(type))
                 throw new InvalidOperationException($"Domainservice {type} cannot be resolved by container, register it before calling map");
 
-            return _dataSource.AddDomainService(longName + "/binary", type);
+            return _dataSource.AddDomainService(GetDomainServiceRoute(type), type);
         }
 
-        public IEndpointConventionBuilder AddDomainService<T>() where T : DomainService
+        public IEndpointConventionBuilder AddDomainService(Type type, string path)
         {
-            return AddDomainService(typeof(T));
+            ArgumentNullException.ThrowIfNull(nameof(type));
+#if NET7_0_OR_GREATER
+            ArgumentNullException.ThrowIfNullOrEmpty(nameof(path));
+#endif
+            if (path.EndsWith('/'))
+                throw new ArgumentException("Path should not end with /", nameof(path));
+
+            if (!_typeIsService.IsService(type))
+                throw new InvalidOperationException($"Domainservice {type} cannot be resolved by container, register it before calling map");
+
+            return _dataSource.AddDomainService(path, type);
+        }
+
+        public IEndpointConventionBuilder AddDomainService<T>(string path) where T : DomainService
+        {
+            return AddDomainService(typeof(T), path);
+        }
+
+        private static string GetDomainServiceRoute(Type type)
+        {
+            // 1. TODO: Look at EnableClientAccessAttribute if we can set route there
+            // 2. Lookup DomainServiceEndpointRoutePatternAttribute in same assembly as DomainService
+            // 3. Lookup DomainServiceEndpointRoutePatternAttribute in startup assembly
+            // 4. Fallback to default (FullName)
+            // - Fallback to FullName
+            EndpointRoutePattern pattern =
+                type.Assembly.GetCustomAttribute<DomainServiceEndpointRoutePatternAttribute>()?.EndpointRoutePattern
+                ?? Assembly.GetEntryAssembly().GetCustomAttribute<DomainServiceEndpointRoutePatternAttribute>()?.EndpointRoutePattern
+                ?? EndpointRoutePattern.FullName;
```

Comment:
Should we switch default now, or keep the default as WCF and do it in a later release ?

Reply:
Will switch to wcf and change in next major release with 85% certainty

Reply:
changed